polar-core/src/error.rs

```diff
@@ -609,6 +619,10 @@ where
     Err(OperationalError::InvalidState { msg }.into())
 }
 
+pub(crate) fn unexpected_value<T>(expected: &'static str, received: Arc<Value>) -> PolarResult<T> {
```

Comment:
Nit: why doesn't this take Term instead of `Arc<Value>`?

Reply:
Over-optimization to avoid cloning the source info; undone in 012ff8e. Not sure if the term we attach to the error will ever actually have context, but this at least sets us up to handle it appropriately if it does (either now or at some point in the future)